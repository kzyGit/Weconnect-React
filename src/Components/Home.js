import React, { Component } from 'react';
import '../Styles/App.css';

import { Link } from 'react-router';

/**
 * Home component
 * Displays welcome message and links to view or create businesses
 */
class Home extends Component {

  render() {
    return (
      <div className="content">
        <div style={{ textAlign: 'center', marginBottom: '12%', marginTop: '12%' }}>

          <h3>Resourceful, Reliable, Individuals - Business connect platform. </h3>
          <h3>You need it, Here you go, <Link to="/businesses" style={{ textDecoration: 'none' }}><font style={{ color: 'rgb(173, 32, 32)', marginLeft: '15px' }}><span className="glyphicon glyphicon-hand-right"></span> Connect *</font></Link></h3><br /><br /><br />

          <Link to="/businesses" id='Homebutton'><button type="submit" className="btn btn-default">View Businesses</button></Link>
          <Link to="/addbusiness" id='Homebutton'><button type="submit" className="btn btn-default">Create Business</button></Link>
        </div>
      </div>
    );
  }
}

export default Home;
