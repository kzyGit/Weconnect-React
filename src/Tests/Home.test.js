import React from 'react';
import { shallow } from 'enzyme';
import Home from '../Components/Home';
import { shallowToJson } from 'enzyme-to-json';

describe('Home component', () => {

    const wrapper = shallow(<Home />);

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('contains divs', () => {
        expect(wrapper.find('div')).toHaveLength(3);
    });
    it('contains Header', () => {
        expect(wrapper.find('Header')).toHaveLength(1);
    });
    it('contains Headers', () => {
        expect(wrapper.find('h3')).toHaveLength(2);
    });
    it('contains Footer', () => {
        expect(wrapper.find('Footer')).toHaveLength(1);
    });

});
