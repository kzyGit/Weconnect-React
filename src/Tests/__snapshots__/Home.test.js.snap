// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders properly 1`] = `
<div
  className="row"
>
  <Header />
  <div
    className="content"
  >
    <div
      style={
        Object {
          "marginBottom": "12%",
          "marginTop": "12%",
          "textAlign": "center",
        }
      }
    >
      <h3>
        Resourceful, Reliable, Individuals - Business connect platform. 
      </h3>
      <h3>
        You need it, Here you go, 
        <Link
          onlyActiveOnIndex={false}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          to="/businesses"
        >
          <font
            style={
              Object {
                "color": "rgb(173, 32, 32)",
                "marginLeft": "15px",
              }
            }
          >
            <span
              className="glyphicon glyphicon-hand-right"
            />
             Connect *
          </font>
        </Link>
      </h3>
      <br />
      <br />
      <br />
      <Link
        id="Homebutton"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/businesses"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          View Businesses
        </button>
      </Link>
      <Link
        id="Homebutton"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/addbusiness"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          Create Business
        </button>
      </Link>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Root component renders properly 1`] = `
<div
  className="row"
>
  <Header />
  <div
    className="content"
  >
    <div
      style={
        Object {
          "marginBottom": "12%",
          "marginTop": "12%",
          "textAlign": "center",
        }
      }
    >
      <h3>
        Resourceful, Reliable, Individuals - Business connect platform. 
      </h3>
      <h3>
        You need it, Here you go, 
        <Link
          onlyActiveOnIndex={false}
          style={
            Object {
              "textDecoration": "none",
            }
          }
          to="/businesses"
        >
          <font
            style={
              Object {
                "color": "rgb(173, 32, 32)",
                "marginLeft": "15px",
              }
            }
          >
            <span
              className="glyphicon glyphicon-hand-right"
            />
             Connect *
          </font>
        </Link>
      </h3>
      <br />
      <br />
      <br />
      <Link
        id="Homebutton"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/businesses"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          View Businesses
        </button>
      </Link>
      <Link
        id="Homebutton"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/addbusiness"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          Create Business
        </button>
      </Link>
    </div>
  </div>
  <Footer />
</div>
`;
